components/support/migration/src/main/java/mozilla/components/support/migration/SearchEngineMigration.kt

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.migration
+
+import android.content.Context
+import mozilla.components.browser.search.SearchEngineManager
+import mozilla.components.support.base.log.logger.Logger
+
+private const val PREF_FENNEC_DEFAULT_ENGINE_KEY = "search.engines.defaultname"
+private const val PREF_FENIX_DEFAULT_ENGINE_KEY = "pref_key_search_engine"
+
+/**
+ * Implementation for migrating the default search engine from Fennec to Fenix.
+ *
+ * This is a "best effort" migration. We will recover the default search engine name from Fennec and
+ * then try to find a matching search engine in the search engine manager for the current Fenix
+ * installation. If we find a match we will update the default search engine to that; otherwise
+ * the regular default search engine for this installation will be used.
+ */
+internal object SearchEngineMigration {
+    private val logger = Logger("SearchEngineMigration")
+
+    /**
+     * Tries to migrate the default search engine from Fennec to Fenix.
+     */
+    suspend fun migrate(
+        context: Context,
+        searchEngineManager: SearchEngineManager
+    ): Result<SearchEngineMigrationResult> {
+        val fennecSearchEngine = retrieveFennecDefaultSearchEngineName(context)
+        if (fennecSearchEngine == null) {
+            logger.debug("Could not locate Fennec search engine preference")
+            return Result.Failure(
+                SearchEngineMigrationException(SearchEngineMigrationResult.Failure.NoDefault))
+        }
+
+        logger.debug("Found search engine from Fennec: $fennecSearchEngine")
+
+        return determineNewDefault(context, searchEngineManager, fennecSearchEngine)
+    }
+
+    private suspend fun determineNewDefault(
+        context: Context,
+        searchEngineManager: SearchEngineManager,
+        name: String
+    ): Result<SearchEngineMigrationResult> {
+        val searchEngines = searchEngineManager.getSearchEnginesAsync(context)
+
+        searchEngines.forEach { engine ->
```

Comment:
maybe a comment here would be useful - if passed-in fennec default doesn't match anything from the search manager, we do nothing - which is equivalent of keeping fenix's default search engine as a "fallback".

Reply:
That is mentioned in the class comment. But I agree, here in context, we could mention that too.